Public/Src/Tools/ContentPlacementAnalysisTools/ContentPlacementAnalysisTools.Core/BxlBuild.cs

```diff
@@ -0,0 +1,384 @@
+    /// </summary>
+    public class BxlArtifact
+    {
+        /// <summary>
+        /// The hash for this artifact
+        /// </summary>
+        public string Hash { get; set; }
+        /// <summary>
+        /// The file that is represented by this artifact
+        /// </summary>
+        public string ReportedFile { get; set; }
+        /// <summary>
+        /// The size of the file associated to this artifact
+        /// </summary>
+        public long ReportedSize { get; set; }
+        /// <summary>
+        /// The number of pips that use this artifact
+        /// </summary>
+        public int NumInputPips => InputPips.Count;
+        /// <summary>
+        /// The data of all the pips that use this artifact
+        /// </summary>
+        public List<BxlPipData> InputPips { get; set; } = new List<BxlPipData>();
+        /// <summary>
+        /// The number of pips that produce this artifact
+        /// </summary>
+        public int NumOutputPips  =>  OutputPips.Count;
+        /// <summary>
+        /// The data of all the pips that produce this artifact
+        /// </summary>
+        public List<BxlPipData> OutputPips { get; set; } = new List<BxlPipData>();
+
+        /// <summary>
+        /// Writes a single artifact to a json stream
+        /// </summary>
+        public void WriteToJsonStream(JsonTextWriter writer)
+        {
+            writer.WriteStartObject();
+            BxlBuild.WriteJsonPropertyToStream(writer, "Hash", Hash);
+            BxlBuild.WriteJsonPropertyToStream(writer, "ReportedFile", ReportedFile);
+            BxlBuild.WriteJsonPropertyToStream(writer, "ReportedSize", ReportedSize);
+            BxlBuild.WriteJsonPropertyToStream(writer, "NumInputPips", NumInputPips);
+            BxlBuild.WriteJsonPropertyToStream(writer, "NumOutputPips", NumOutputPips);
+            WritePipsToJsonStream(writer, "InputPips", InputPips);
+            WritePipsToJsonStream(writer, "OutputPips", OutputPips);
+            writer.WriteEndObject();
+            // done...
+        }
+
+        private void WritePipsToJsonStream(JsonTextWriter writer, string fieldName, List<BxlPipData> pips)
+        {
+            writer.WritePropertyName(fieldName);
+            writer.WriteStartArray();
+            foreach (var pip in pips)
+            {
+                pip.WriteToJsonStream(writer);
+            }
+            writer.WriteEndArray();
+        }
+
+        /// <summary>
+        /// Utility method to create a bxl pip artifact from a json file. The attrs are read in
+        /// the same order they are writen
+        /// </summary>
+        public static BxlArtifact ReadFromJsonStream(JsonTextReader reader)
+        {
+            var output = new BxlArtifact();
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonToken.StartObject)
+                {
+                    // here does this guy starts. This guy has several properties that we will read,

```

Comment:
Nit: Same as above, change "this guy" to "The reader" or something along that sort

Reply:
There are a few more places where you can change it as well.